compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -6481,6 +6544,14 @@ object Types extends TypeUtils {
           if (underlying.isExactlyNothing) underlying
           else tp.derivedAnnotatedType(underlying, annot)
       }
+    override protected def derivedFlexibleType(tp: FlexibleType, underlying: Type): Type =
+      underlying match {
+        case Range(lo, hi) =>
+          range(tp.derivedFlexibleType(lo), tp.derivedFlexibleType(hi))
+        case _ =>
+          if (underlying.isExactlyNothing) underlying
```

Comment:
Should this logic not be part of the `derivedFlexibleType` in `FlexibleType` instead?

Reply:
Now the returned range does reflect the underlying "range" of a flexible type.